frankenphp.c

```diff
@@ -661,9 +754,24 @@ static void frankenphp_register_variables(zval *track_vars_array) {
   /* https://www.php.net/manual/en/reserved.variables.server.php */
 
   /* In CGI mode, we consider the environment to be a part of the server
-   * variables
+   * variables.
    */
-  php_import_environment_variables(track_vars_array);
+  struct go_getfullenv_return full_env = go_getfullenv(thread_index);
```

Comment:
Have you also tried this approach with variable_order `EGPCS`? I think we also need to add the import logic to the
`sapi_module_struct frankenphp_sapi_module` at `/* getenv */`

Reply:
Here's how [FPM does it](https://github.com/php/php-src/blob/b9a64c5e344e2db2ebac0294a423d9e5d04764f7/sapi/fpm/fpm/fpm_main.c#L805). It looks more like it's overriding `getenv` though, and not actually creating the `$_ENV` global.

Reply:
I think we would need to override the `auto_global` import function [here](https://github.com/php/php-src/blob/b9a64c5e344e2db2ebac0294a423d9e5d04764f7/main/php_variables.c#L977) or somehow make sure it only gets called once.

Reply:
Looking through `fpm_main.c` again, I think it happens [here ](https://github.com/php/php-src/blob/b9a64c5e344e2db2ebac0294a423d9e5d04764f7/sapi/fpm/fpm/fpm_main.c#L1804)by replacing a function pointer

Reply:
> Here's how [FPM does it](https://github.com/php/php-src/blob/b9a64c5e344e2db2ebac0294a423d9e5d04764f7/sapi/fpm/fpm/fpm_main.c#L805). It looks more like it's overriding getenv though, and not actually creating the $_ENV global.

There is a sapi hook for `getenv` but not `putenv` and it half-way works "correctly." `getenv` has a third `local` parameter that allows you to bypass the sapi hook and use the TSRM implementation. We don't want this to happen. Thus, we have to replace the entire function.

> Have you also tried this approach with variable_order EGPCS?

Seems this case isn't tested. Added failing tests and will look into a solution.